test/integration/features/step_definitions/environments-steps.mjs

```diff
@@ -75,7 +75,11 @@ Given('an environment exists with a {string} branches deployment branch policy',
   )
 
   if (policyType === 'custom') {
-    this.customBranches = any.listOf(() => ({ name: any.word(), id: any.integer() }))
+    this.customBranches = any.listOf(() => ({
+      name: any.word(),
+      id: any.integer(),
+      type: any.fromList(['branch', 'tag'])
```

Comment:
i appreciate that you tracked down how to accomplish this and i think this is totally valid for the scenario that defines `type`, but i think this change now results in not having a scenario tested for the option to not include type, doesnt it. it seems like we probably should define an additional scenario for including type in https://github.com/repository-settings/app/blob/master/test/integration/features/environments.feature

i also wonder if the inclusion of a type that doesnt match the default requires any additional care to avoid unnecessary update requests when the intended value matches what is in the api. could you verify that comparing an existing rule in the api avoids extra calls when the type is and is not present in the settings.yml? if that requires additional logic to avoid extra calls, we may need test scenarios for those cases as well

Reply:
Actually, we always specify `type` for API calls.
https://github.com/repository-settings/app/pull/910/files#diff-d3173e671826d864f026b433955eeef4008090a5fc6c3111f8ec2e3977e2bc4dR156

So, the test is correct and does not dismiss the case without type.